esp/services/WsDeploy/WsDeployService.hpp

```diff
@@ -512,7 +512,7 @@ class CWsDeployFileInfo : public CInterface, implements IConfigFileObserver
         m_pGraphXml.clear();
         m_pNavTree.clear();
 
-        Owned<IEnvironmentFactory> factory = getEnvironmentFactory();       
+        Owned<IEnvironmentFactory> factory = getEnvironmentFactory(true);
```

Comment:
There appears to be an independent subscription / update service in WsDeployService, but I'm not sure why ?
i.e. CWsDeployFileInfo::CSdsSubscription, which ends up calling environmentUpdated() , but the built in validateCache mechanism will be doing the same afaics...

Reply:
I saw the following code inside the environmentUpdated(), Could that be why the independent subscription / update service is needed in WsDeployService?
'''
        m_pEnvXml.clear();
        m_pGraphXml.clear();
        m_pNavTree.clear();
'''

Reply:
I guess so, shame it's duplicating the mechanism though.
But it means it should not be calling getEnvironmentFactory with true, as this is installing it's own update mechanism.